###Assignment-3
Compare Linear Regression Models with Default vs. Custom Parameters for Earthquake Magnitude Prediction
###Objective
Train and analyze two linear regression models to predict earthquake Magnitude: one with default parameters and one with custom-tuned parameters. Compare their performances and interpret findings, and last provide a summary report.

In [11]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [12]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error,r2_score

In [13]:
df=pd.read_csv('/content/sample_data/preprocessed_earthquake_data.csv')
df.head()

Latitude  Longitude        Type     Depth  Magnitude Magnitude Type  \
0  0.583377   0.844368  Earthquake  0.495984   0.277668             MW   
1  0.006109   0.698849  Earthquake  0.075272  -0.195082             MW   
2 -0.739162  -1.701962  Earthquake -0.413928   0.750418             MW   
3 -2.017599  -0.503524  Earthquake -0.454694  -0.195082             MW   
4  0.340688   0.691479  Earthquake -0.454694  -0.195082             MW   

   Root Mean Square  Source     Status      Year  ...  Source_ISCGEM  \
0         -0.103839  ISCGEM  Automatic -1.915523  ...            1.0   
1         -0.103839  ISCGEM  Automatic -1.915523  ...            1.0   
2         -0.103839  ISCGEM  Automatic -1.915523  ...            1.0   
3         -0.103839  ISCGEM  Automatic -1.915523  ...            1.0   
4         -0.103839  ISCGEM  Automatic -1.915523  ...            1.0   

   Source_ISCGEMSUP  Source_NC  Source_NN  Source_OFFICIAL  Source_PR  \
0               0.0        0.0        0.0              0.0        0.0   
1               0.0        0.0        0.0              0.0        0.0   
2               0.0        0.0        0.0              0.0        0.0   
3               0.0        0.0        0.0              0.0        0.0   
4               0.0        0.0        0.0              0.0        0.0   

   Source_SE  Source_US  Source_UW  Status_Reviewed  
0        0.0        0.0        0.0              0.0  
1        0.0        0.0        0.0              0.0  
2        0.0        0.0        0.0              0.0  
3        0.0        0.0        0.0              0.0  
4        0.0        0.0        0.0              0.0  

[5 rows x 40 columns]

In [14]:
drop_columns=['Type','Magnitude Type','Status','Source','Year','Day']
X=df.drop(columns=drop_columns+['Magnitude'])
y=df['Magnitude']

In [15]:
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=42)

###DEFAULT LINEAR REGRESSION

In [17]:
X_train=X_train.fillna(X_train.mean())
X_test=X_test.fillna(X_train.mean())

In [18]:
lin_reg_default=LinearRegression()
lin_reg_default.fit(X_train,y_train)

LinearRegression()

In [24]:
y_pred_default=lin_reg_default.predict(X_test)

In [25]:
mse_default=mean_squared_error(y_test,y_pred_default)
r2_default=r2_score(y_test,y_pred_default)

In [ ]:
print("DEFAULT LINEAR REGRESSION")
print("MSE:",mse_default)
print("R2:",r2_default)

DEFAULT LINEAR REGRESSION
MSE: 0.9465995619040072
R2: 0.08880386062845225


### CUSTOM LINEAR REGRESSION

In [19]:
lin_reg_custom=LinearRegression(fit_intercept=False,copy_X=False)
lin_reg_custom.fit(X_train,y_train)

LinearRegression(copy_X=False, fit_intercept=False)

In [20]:
y_pred_custom=lin_reg_custom.predict(X_test)

In [21]:
mse_custom=mean_squared_error(y_test,y_pred_custom)
r2_custom=r2_score(y_test,y_pred_custom)

In [27]:
print("CUSTOM LINEAR REGRESSION")
print("MSE:",mse_custom)
print("R2:",r2_custom)

CUSTOM LINEAR REGRESSION
MSE: 1.0380649111790303
R2: 0.04594154232899228


### COMPARISON

In [26]:
print("\n---Comparison---")
print(f"Default -> MSE: {mse_default:.3f}, R2: {r2_default:.3f}")
print(f"Custom -> MSE: {mse_custom:.3f}, R2: {r2_custom:.3f}")


---Comparison---
Default -> MSE: 1.038, R2: 0.046
Custom -> MSE: 1.038, R2: 0.046


### SUMMARY

In this assignment ,we compared two Linear Regression models for predicting earthquake magnitude.One using default parameters and other with a custom parameters.Both models were trained and evaluated on the same dataset.The performance was measured using Mean Squared Error(MSE) and Coefficient of determination(R2).The results show that the default model achieved on MSE of 1.038 and R2 of 0.046,while the custom model produced the identical values .This indicates that in this case modifying the model parameters did not improve predictive performance.The Linear Regression prediction is not improved because of the non-linearity of the earthquake data.Overall,the comparison exibit that more advanced techniques are required other than Linear Regression.